trie/trie.go

```diff
@@ -1026,11 +1030,13 @@ func (t *Trie) resolveShadowNode(epoch types.StateEpoch, cur node, prefix []byte
 		return errors.New("cannot resolve shadow node")
 	}
 
-	switch n := cur.(type) {
+	switch n := origin.(type) {
 	case *shortNode:
+		n.setEpoch(epoch)
 		n.shadowNode.ShadowHash = nil
 		return t.resolveShadowNode(epoch, n.Val, append(prefix, n.Key...))
 	case *fullNode:
+		n.setEpoch(epoch)
```

Comment:
in a case where resolveShadowNode fails, the epoch of nodes along the path until the failed node will still be updated (i.e. `n.setEpoch(epoch)`)

will it be better if when only resolveShadowNode succeeds, then we recursively update the epoch of each node from children to its parent?

Reply:
I think the order is ok. When the child's resolveShadowNode failed and does not update its epoch, but the node has been solved, it could use to delete or update, it will use epoch0 as default.